src/main/java/org/sagebionetworks/web/client/security/AuthenticationControllerImpl.java

```diff
@@ -82,6 +90,31 @@ public void onSuccess(UserSessionData userSessionData) {
 					cookies.setCookie(CookieKeys.USER_LOGIN_TOKEN, userSessionData.getSession().getSessionToken(), tomorrow);
 					currentUser = userSessionData;
 					callback.onSuccess(userSessionData);
+					
+					// Attempt to get UserBundle information for user being set

```

Comment:
Clear user bundle before any remote async call.  And only callback after finishing init of userbundle (callback.onSuccess(userSessionData);).  Change login call to return user session data and user bundle (unless usersessiondata is a portal transfer object, then just add a UserBundle field to it).